semgrep-core/src/core/il/IL_lvalue_helpers.ml

```diff
@@ -90,13 +92,39 @@ let rlvals_of_instr x =
 (* API *)
 (*****************************************************************************)
 
+let dotted_lvars_of_lval = function
+  | { base = Var name; offset } -> (
+      let id, tok = name.ident in
+      let str_of_name name = Common.spf "%s-%d" (fst name.ident) name.sid in
+      let dot_strs =
+        List.fold_right
+          (fun o s ->
+            match (s, o) with
+            | Some (s :: ss), Dot name ->
+                let x = s ^ "." ^ str_of_name name in
+                Some (x :: s :: ss)
+            | Some [], Dot name -> Some [ "." ^ str_of_name name ]
+            | _ -> None)
+          offset (Some [])
+      in
+      match dot_strs with
+      | None
+      | Some [] ->
+          []
+      | Some dot_strs ->
+          let add_dots dots = { name with ident = (id ^ ";" ^ dots, tok) } in
+          Common.map add_dots dot_strs)
+  | _ -> []
+
 let lvar_of_instr_opt x =
   let* lval = lval_of_instr_opt x in
-  match lval.base with
-  | Var n -> Some n
-  | VarSpecial _
-  | Mem _ ->
-      None
+  match lval with
+  | { base = Var name; _ } -> (
+      let dots = dotted_lvars_of_lval lval in
+      match dots with
+      | dotted_name :: _ -> Some (name, dotted_name, dots)
```

Comment:
This may be obvious later on as I review the other changes, but right here I don't understand why we need to return `dotted_name` separately here. Isn't it already included in `dots`? We could have a short explanation here.

Reply:
It was in fact not needed